### PR TITLE
Crab (New) - Advanced salvaging shuttle

### PR DESCRIPTION
**Pull request by RealIHaveANameOfficial**
## About the PR

This adds the Crab, which is an advanced auto-salvaging shuttle. Rather than have to go through the manual labor of mining asteroids, now you can use the power of technology to do the hard work for you. All you have to do is collect and enjoy. With our patented Claw Grip containment system, you'll ensure no rogue asteroids evade your grasp. Tentatively priced at 50,500 spesos.

## Why / Balance

Thought it was a cool idea and I'm still on my crab kick.
## Technical details

The Crab has a modest medical section, a captain's quarters, a salvager's room, a bathroom, a 2 core AME, and a rudimentary atmospheric system.
The feature here are the arrays of particle emitters, all linked up to a button, with the goal of mining the hell out of any asteroid you can grab.
That leads us to the second feature, the particle containment generators. These are included to ensure any debris stays locked to where it needs to be.
## Media

image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: The Crab, an advanced auto-mining vessel.

### CODE REVIEW

File: Content.Shared/Singularity/Components/ContainmentFieldGeneratorComponent.cs [hunk 1]
```diff
@@ -59,7 +59,7 @@ public int PowerBuffer
     /// How many tiles should this field check before giving up?
     /// </summary>
     [DataField("maxLength")]
-    public float MaxLength = 8F;
+    public float MaxLength = 16F;
```
Comment (dvir001) [inline Content.Shared/Singularity/Components/ContainmentFieldGeneratorComponent.cs:62]: Dont do this in the same PR
Also need to add a note about it being frontier change

# Frontier

Comment (RealIHaveANameOfficial) [reply Content.Shared/Singularity/Components/ContainmentFieldGeneratorComponent.cs:62]: what about a frontier change? Also, noted, I will make another PR for containment field generators

**Comment by Cheackraze** [comment]
i think the idea is clever, but thats a lot of emitters available on a ship and I'm concerned with it being purchased just to troll people with. It also has a ton of lighting, you should trim the amount of available lights in it

**Comment by Cheackraze** [comment]
if it had maybe 4-5 emitters total i could maybe allow it

**Comment by MagnusCrowe** [comment]
Going to close this for now.  Re-open it if you wish to fix the merge issues.

**Pull request closed without merging**